kubespawner/spawner.py

```diff
@@ -2285,6 +2285,8 @@ async def get_pod_manifest(self):
         annotations = self._build_common_annotations(
             self._expand_all(self.extra_annotations)
         )
+        # Unconditionally put servername on the pods
+        annotations['hub.jupyter.org/servername'] = self.name
```

Comment:
It looks like the unconditional assignment following right after calling _build_common_annotations should only be relevant if self.extra_annotations were declaring another `hub.jupyter.org/servername`, which would have a blank string value or similar.

```python
        annotations = self._build_common_annotations(
            self._expand_all(self.extra_annotations)
        )
        # Unconditionally put servername on the pods
        annotations['hub.jupyter.org/servername'] = self.name
```

I wonder if its really a kubespawner issue, or z2jh / end-user issue based on this.